src/services/tasks/apiRequests/types.ts

```diff
@@ -0,0 +1,88 @@
+import { ApiRequestLogsService } from './logs/types';
+import { ObjectId, OptionsWithRql, PagedResultWithPager } from '../../types';
+import { FindAllIterator } from '../../helpers';
+import { ApiFunctionRequestMethod } from '../api/types';
+
+export interface ApiRequestService {
+  /**
+   * ## Retrieve a paged list of API Requests
+   *
+   * **Global Permissions:**
+   * - `VIEW_API_FUNCTION_REQUESTS` - Allows a user to view API Requests
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns A paged list of API Requests {@link PagedResultWithPager PagedResultWithPager<ApiRequest>}
+   */
+  find(options?: OptionsWithRql): Promise<PagedResultWithPager<ApiRequest>>;
+
+  /**
+   * ## Retrieve a list of all API Requests
+   *
+   * **Global Permissions:**
+   * - `VIEW_API_FUNCTION_REQUESTS` - Allows a user to view API Requests
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns An array of API Requests {@link ApiRequest ApiRequest[]}
+   * @throws {@link Error} Do not pass in limit operator with findAll
+   */
+  findAll(options?: OptionsWithRql): Promise<ApiRequest[]>;
+
+  /**
+   * ## Retrieve a paged list of API Requests
+   *
+   * **Global Permissions:**
+   * - `VIEW_API_FUNCTION_REQUESTS` - Allows a user to view API Requests
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns An iterator for the queried API Requests {@link FindAllIterator FindAllIterator<ApiRequest>}
+   */
+  findAllIterator(options?: OptionsWithRql): FindAllIterator<ApiRequest>;
+
+  /**
+   * ## Retrieve the first queried API Request
+   *
+   * **Global Permissions:**
+   * - `VIEW_API_FUNCTION_REQUESTS` - Allows a user to view API Requests
+   * @param options {@link OptionsWithRql} - Add filters to the requested list
+   * @returns The first element of the queried API Requests {@link ApiRequest}
+   */
+  findFirst(options?: OptionsWithRql): Promise<ApiRequest>;
+  logs: ApiRequestLogsService;
+}
+
+export interface ApiRequest {
+  /** The Extra Horizon document id */
+  id: ObjectId;
+  /** The unique identifier amongst all Functions */
+  functionName: string;
+  /**  The HTTP method used to target the API function. */
+  method: ApiFunctionRequestMethod;
+  /** The path portion of the URL that comes after the Function name. */
+  path: string;
+  /** The Extra Horizon user id */
+  userId?: ObjectId;
+  /** The Extra Horizon application id */
+  applicationId?: ObjectId;
```

Comment:
(minor) I like how we described this for the request headers:
https://docs.extrahorizon.com/extrahorizon/services/automation/task-service/api-functions#headers